mmcv/ops/nms.py

```diff
@@ -285,10 +287,12 @@ def batched_nms(boxes, scores, idxs, nms_cfg, class_agnostic=False):
     nms_op = eval(nms_type)
 
     split_thr = nms_cfg_.pop('split_thr', 10000)
-    if boxes_for_nms.shape[0] < split_thr:
+    # Won't split to multiple nms nodes when exporting to onnx
+    if boxes_for_nms.shape[0] < split_thr or torch.onnx.is_in_onnx_export():
         dets, keep = nms_op(boxes_for_nms, scores, **nms_cfg_)
         boxes = boxes[keep]
-        scores = dets[:, -1]
+        # Keep tracing when export to ONNX.
+        scores = scores[keep]
```

Comment:
This will cause bug when using soft nms as soft nms will modify the scores and we should use `det[:, -1]`.

Reply:
`-1` indexing works abnormally in TensorRT, changed to `4`